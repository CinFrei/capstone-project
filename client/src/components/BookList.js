import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

import BookCard from './BookCard'

/** @component */
export default BookList

BookList.propTypes = {
  listName: PropTypes.string,
  newBooks: PropTypes.array,
  showDetail: PropTypes.func,
}

function BookList({ listName, newBooks, showDetail }) {
  return (
    <BookListWrapper>
      <h2>{listName}</h2>
      {newBooks.map((newBook) => (
        <BookCard
          author={newBook.volumeInfo.authors}
          buttonName="Mehr Ã¼ber's Buch"
          cover={newBook.volumeInfo.imageLinks.thumbnail}
          onCardButtonClick={() => showDetail(newBook.id)}
          key={newBook.id}
          title={newBook.volumeInfo.title}
        />
      ))}
    </BookListWrapper>
  )
}

const BookListWrapper = styled.ol`
  display: grid;
  grid-gap: 10px;
  grid-template-rows: 1fr auto;
  padding: 20px;

  h2 {
    margin: 0;
    text-align: left;
  }
`
