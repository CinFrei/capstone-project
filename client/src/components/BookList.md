Basic BookList:

```js
<BookList
newBooks={{
    books: {
    1: { author: '["Paulchen Panter", "Peter Pan"]',
      buttonName: 'Buch entfernen',
      cover: './public/styleguide/sample-cover.jpg"',
      title: 'Pauls Rosa Welt',},
    2: { author: '["Paulchen Panter", "Peter Pan"]',
      buttonName: 'Buch entfernen',
      cover: './public/styleguide/sample-cover.jpg"',
      title: 'Pauls Rosa Welt',},
    },
  }
  deleteBook={() => {}}
  listName="BÃ¼cherregal"

/>
```
