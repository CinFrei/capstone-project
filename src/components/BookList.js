import BookCard from './BookCard'
import PropTypes from 'prop-types'
import styled from 'styled-components/macro'

/** @component */
export default BookList

BookList.propTypes = {
  listName: PropTypes.string,
  newBooks: PropTypes.array,
}

function BookList({ newBooks, listName, deleteBook }) {
  return (
    <BookListWrapper>
      <h2>{listName}</h2>
      {newBooks.map((newBook) => (
        <BookCard
          author={newBook.volumeInfo.authors}
          buttonName="Buch entfernen."
          cover={newBook.volumeInfo.imageLinks.thumbnail}
          key={newBook.id}
          onCardBtnClick={() => deleteBook(newBook.id)}
          title={newBook.volumeInfo.title}
        />
      ))}
    </BookListWrapper>
  )
}

const BookListWrapper = styled.section`
  display: grid;
  grid-gap: 10px;
  grid-template-rows: 1fr auto;
  padding: 10px;

  h2 {
    margin: 0;
    text-align: left;
  }
`
