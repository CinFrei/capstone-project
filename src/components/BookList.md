Basic BookList:

```js
<BookList listName="BÃ¼cherregal" />
```