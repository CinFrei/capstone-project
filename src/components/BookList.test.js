import BookList from './BookList'
import { render } from '@testing-library/react'

describe('BookList', () => {
  it('renders correctly', () => {
    const { container } = render(
      <BookList
        listName="BÃ¼cherregal"
        newBooks={[
          {
            author: 'Paulchen Panter',
            description: 'Rosa ist die Beste Farbe',
            id: '007',
            title: 'Pauls Rosa Welt',
          },
        ]}
      />
    )
    expect(container.firstChild).toMatchSnapshot()
  })
})
