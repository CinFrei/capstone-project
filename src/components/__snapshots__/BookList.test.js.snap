// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BookList renders correctly 1`] = `
.c1 {
  position: relative;
  min-width: 0;
  display: grid;
  grid-template-columns: 40% 60%;
  grid-gap: 10px;
  padding: 10px;
  background: var(--primary-white);
  background: var(--primary-background-gradient);
  border: 2px solid var(--dark-blue);
  border-radius: 3px;
}

.c1 h3 {
  margin: 10px;
}

.c1 h4 {
  display: inline;
  margin: 10px;
}

.c1 img {
  width: 100%;
}

.c0 {
  display: grid;
  grid-template-rows: 1fr auto;
  grid-gap: 10px;
  padding: 10px;
}

.c0 h2 {
  text-align: left;
  margin: 0;
}

<section
  class="c0"
>
  <h2>
    Bücherregal
  </h2>
  <div
    class="c1"
  >
    <img
      alt=""
      src="https://books.google.de/books/content?id=8GWIBguHHVQC&printsec=frontcover&img=1&zoom=5&edge=curl&imgtk=AFLRE73YwZT_dnPkne8qiPiU9yJIDrEOsm2_kyUvaNqumPcVbLozARgLWE5Ca7DaY4aR0dCZLx4zUNheZP8mXEC2bl1u_pv37Dk-61YPSd7s7budgbhKWGPyn4tR4RTWHtf7mDHbttCI"
    />
    <div>
      <h3>
        Der letzte Wunsch
      </h3>
      <span>
        von
        <h4>
          Andrzej Sapkowski
        </h4>
      </span>
      <p>
        Der Hype um den Hexer geht weiter.
      </p>
    </div>
  </div>
  <div
    class="c1"
  >
    <img
      alt=""
      src="https://books.google.de/books/content?id=BJoRE_AvtNQC&printsec=frontcover&img=1&zoom=5&edge=curl&imgtk=AFLRE73PqAB6J0LAYdo51QPLGGRdWB4aYx4sT7tl3MaAXNxhQnnzZV4bXs1go3LWjTvhkrzMmQHxBGC9WDn04E5GKWkZECBUM3ksot6WUeyqvylLJIUjMzh7EQRp-Y3WGbQsXIR0Jj7B"
    />
    <div>
      <h3>
        Die Zeit der Verachtung
      </h3>
      <span>
        von
        <h4>
          Andrzej Sapkowski
        </h4>
      </span>
      <p>
        Die Hexer-Saga 2 in der opulenten Fan-Edition Ciri wird von allen Seiten gejagt.
      </p>
    </div>
  </div>
  <div
    class="c1"
  >
    <img
      alt=""
      src="https://books.google.de/books/content?id=zh_wBgAAQBAJ&printsec=frontcover&img=1&zoom=5&edge=curl&imgtk=AFLRE73KmbXI1-yI2jSoUsOqLhsAoqOn7rRvgREihvb-YV2tGH9yDL1bElbePzp7OVBUhsNZcN5hOa2RJ5Pd9kXh00A5vf8tpKs5bR3jYYFMOG21-zXSwv5G5SQoJqCxOkvERPM2Mi2I"
    />
    <div>
      <h3>
        Die Zeit des Sturms
      </h3>
      <span>
        von
        <h4>
          Andrzej Sapkowski
        </h4>
      </span>
      <p>
        Die Vorgeschichte 2 in der opulenten Fan-Edition Das Königreich Kerack wird von Kämpfen um den Thron erschüttert.
      </p>
    </div>
  </div>
  <div
    class="c1"
  >
    <img
      alt=""
      src="https://books.google.de/books/content?id=0zQQkc8uHXMC&printsec=frontcover&img=1&zoom=5&edge=curl&imgtk=AFLRE71nrLI7QflNydctB5_NxDqjtWAeojCAQTH3-KWEZk_8b1j6mGETIEZSKAjpT7zas_3IhuOh9mGG099CzVQgzK2-eRY9ff0ENS_9vxtbe7u6exBCEvsj-rap3i1mtMytcdqO7SJ-"
    />
    <div>
      <h3>
        Das Schwert der Vorsehung
      </h3>
      <span>
        von
        <h4>
          Andrzej Sapkowski
        </h4>
      </span>
      <p>
        Untiere, Dämonen, Spannung und Witz Was soll man von dir halten - ein Hexer, der jeden zweiten Tag ein einträgliches Angebot ausschlägt?
      </p>
    </div>
  </div>
  <div
    class="c1"
  >
    <img
      alt=""
      src="https://books.google.de/books/content?id=tWweq1PC8goC&printsec=frontcover&img=1&zoom=5&edge=curl&imgtk=AFLRE73z1Hge8UFkxcgikuZ9oriB-8Uafwu09e0tFMQMaKxZ1ukz0x9YgQn3A_TYwF8a0tnoQEdhha-jpiMFXMMktvd8_wkUp62IDpFhmYYSv18ktjI37G8Dwjejg4gWP1zYiOVVrDSg"
    />
    <div>
      <h3>
        Die Dame vom See
      </h3>
      <span>
        von
        <h4>
          Andrzej Sapkowski
        </h4>
      </span>
      <p>
        Dort trifft sie auf einen Ritter, der sie für die Dame vom See hält. Eine letzte große Schlacht wird das Schicksal von Ciri und Geralt erfüllen: Wird sich die Prophezeiung bewahrheiten?
      </p>
    </div>
  </div>
  <div
    class="c1"
  >
    <img
      alt=""
      src="https://books.google.de/books/content?id=CV9Xv4mM9h4C&printsec=frontcover&img=1&zoom=5&edge=curl&imgtk=AFLRE70k4RkDc336byIxXbEWy1gqS7MSn9VtfgF5Hd02VOfpRtqYDzr3M6RY6itss4RbaTuM2MBHFJS2c3SjhixbXZODaRn-fedx79r51sBa434wlV1MAjcyXLETlt7XQ9N7XzdR4MvE"
    />
    <div>
      <h3>
        Feuertaufe
      </h3>
      <span>
        von
        <h4>
          Andrzej Sapkowski
        </h4>
      </span>
      <p>
        In Nilfgaard wird die Verlobung des Kaisers mit Cirilla, Thronerbin von Cintra, proklamiert.
      </p>
    </div>
  </div>
  <div
    class="c1"
  >
    <img
      alt=""
      src="https://i.gr-assets.com/images/S/compressed.photo.goodreads.com/books/1347751626l/1422251.jpg"
    />
    <div>
      <h3>
        Die zwölf Königreiche
      </h3>
      <span>
        von
        <h4>
          Jūni Kokuki
        </h4>
      </span>
      <p>
        Die Oberschülerin Yōko wird durch eine Art Dimensionsstrudel ins Land der „12 Königreiche“ gebracht.
      </p>
    </div>
  </div>
</section>
`;
